src/autoimport/model.py

```diff
@@ -52,6 +52,8 @@ def __init__(
         self.code: List[str] = []
         self.config: Dict[str, Any] = config if config else {}
         self._trailing_newline = False
+        self._affect_indented_imports = self.config.get("affect_indented_imports", True)
```

Comment:
In what case would you be interested in not changing the indented imports?

Reply:
Answering your reply below:

There are many reasons to move all the imports to the top:

* [It's stated by PEP8](https://www.python.org/dev/peps/pep-0008/#imports)

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

* [It's more efficient](https://stackoverflow.com/questions/128478/should-import-statements-always-be-at-the-top-of-a-module)

If you want to load some modules only if a function is run, it's best to use lazy loading. Some links of information about it:

* https://files.bemusement.org/talks/OSDC2008-FastPython/
* https://stackoverflow.com/questions/4177735/best-practice-for-lazy-loading-python-modules
* https://snarky.ca/lazy-importing-in-python-3-7/
* https://levelup.gitconnected.com/python-trick-lazy-module-loading-df9b9dc111af

I planned to do an article in the [blue book](https://lyz-code.github.io/blue-book/) about it, but I haven't found the time yet

Reply:
> There are many reasons to move all the imports to the top

That's why it's the majority of cases; but if the import wasn't written on the top, perhaps there's a reason for it. Lazy loading is just one of those possible reasons. And even for that reason, as suggested in one of the links:

>  * https://stackoverflow.com/questions/4177735/best-practice-for-lazy-loading-python-modules

simply importing a module within function is enough for it.

The only opposite use-case I can think of, is writing non-common imports near the code and get `autoimport` to move them to the top. Which is why it at least makes sense to make it configurable. Although that sounds more like an `autoflake`'s job rather than `autoimport`.

Reply:
But if you configure it in your `pyproject.toml` none of the indented imports of the project would be moved to the top. I'd rather use the `# noqa: autoimport` on those particular lines instead of having it as a configuration.

But maybe I'm being too obtuse on this, what do you think @Jasha10 @muriel?

Reply:
I feel conflicted about this.
Here is my datapoint:

Before I started using autoimport, I would occasionally put imports inside of functions or in the middle of a module. My main reasons for wanting a given import to be near its call-site were enhanced readability and greater ease of refactoring.

A few examples of indented imports I've seen in the wild:
- within an [`if TYPE_CHECKING`](https://mypy.readthedocs.io/en/stable/runtime_troubles.html?highlight=TYPE_CHECKING#typing-type-checking) block
- branching based on python version e.g. `importlib.resources` for python3.7 vs `importlib_resources` for python3.6.

It's good to know that `# noqa: autoimport` is an option for granular control in these cases.

---

In my mind, this is a question of how opinionated we want autoimport to be -- should it follow the mold of `black`, which has very few configuration options? The other extreme would be to follow `flake8`, which allows granular configuration of individual behaviors on a per-scope basis.

Pros of the granular-control `flake8` model:
- gives users more latitude to selectively "break the rules"
- May encourage user contribution of new config options

Pros of the opinionated `black` model:
- Using highly opinionated tools can give a sense of security and predictability.
- Easier to implement and maintain. This is especially true given that `black` (and `autoimport`) deal with source-code transformations, so enabling/disabling one behavior may have nontrivial interaction with other behaviors. Contrast this with `flake8`, where disabling one specific error should have no effect on whether other errors are caught.

Reply:
Thank you for your comments @Jasha10, you helped me clear up my mind. I think `autoimport` should support the two cases you mentioned of indented imports, in fact, the `if TYPE_CHECKING` it's already supported. The other one is more difficult to implement so as for now we'll have to use the `noqa` comment.

Answering the question on how opinionated we want it to be, I'll go for the `black` path with the idea to maximize the maintainability. I'd agree to add user configurations when they raise them and they don't break good coding practices. 

So I'd say that in this case, if you want to have indented imports that are not moved to the top, you should use the `noqa` comment, sorry @HoverHell :(